Running a more comprehensive set of tests is revealing errors for the `bool` dtype; for example:
```
LaxBackedNumpyTests.testPadSymmetricAndReflect_shape=bool[1,4]_mode=symmetric_pad_width=((1, 2),)_reflect_type=odd

Traceback (most recent call last):
  File "/build/work/5af997c0288660df34bdcf33f52a84c4f9b3/google3/runfiles/google3/third_party/py/absl/testing/parameterized.py", line 299, in bound_param_test
    return test_method(self, **testcase_params)
  File "/build/work/5af997c0288660df34bdcf33f52a84c4f9b3/google3/runfiles/google3/third_party/py/jax/tests/lax_numpy_test.py", line 1418, in testPadSymmetricAndReflect
    tol={np.float32: 1e-3, np.complex64: 1e-3})
  File "/build/work/5af997c0288660df34bdcf33f52a84c4f9b3/google3/runfiles/google3/third_party/py/jax/test_util.py", line 900, in _CheckAgainstNumpy
    canonicalize_dtypes=canonicalize_dtypes)
  File "/build/work/5af997c0288660df34bdcf33f52a84c4f9b3/google3/runfiles/google3/third_party/py/jax/test_util.py", line 827, in assertAllClose
    self.assertDtypesMatch(x, y, canonicalize_dtypes=canonicalize_dtypes)
  File "/build/work/5af997c0288660df34bdcf33f52a84c4f9b3/google3/runfiles/google3/third_party/py/jax/test_util.py", line 807, in assertDtypesMatch
    self.assertEqual(_dtype(x), _dtype(y))
AssertionError: dtype('int64') != dtype('bool')
```

I think it's probably reasonable to test this over `number_dtypes` rather than `all_dtypes`

--- a/jax/_src/numpy/lax_numpy.py
+++ b/jax/_src/numpy/lax_numpy.py
@@ -2349,7 +2349,6 @@ def _pad_symmetric_or_reflect(array, pad_width, mode, reflect_type):
 
         if reflect_type == 'odd':
           x = 2 * edge - x
-          x = x.astype(array.dtype) # Unexpected type conversion might happened
           if n > 1:
             if before:
               edge = lax.slice_in_dim(x, 0, 1, axis=i)

--- a/tests/lax_numpy_test.py
+++ b/tests/lax_numpy_test.py
@@ -1405,7 +1405,7 @@ class LaxBackedNumpyTests(jtu.JaxTestCase):
       for reflect_type in ['even', 'odd']
       if (pad_width != () and
           # following types lack precision when calculating odd values
-          (reflect_type != 'odd' or dtype not in [np.float16, jnp.bfloat16]))))
+          (reflect_type != 'odd' or dtype not in [np.bool_, np.float16, jnp.bfloat16]))))
   def testPadSymmetricAndReflect(self, shape, dtype, mode, pad_width, reflect_type):
     rng = jtu.rand_default(self.rng())
     args_maker = lambda: [rng(shape, dtype)]

